dev/release-artifacts.py

```diff
@@ -114,28 +125,29 @@ def make_pysrc_wheel(
             orig_pyproj_lines = f.read()
         with open("tests/buildkite/remove_nccl_dep.patch", "r") as f:
             patch_lines = f.read()
-        subprocess.run(["patch", "-p0"], input=patch_lines, text=True)
-
-    with DirectoryExcursion(os.path.join(ROOT, "python-package")):
-        subprocess.check_call(["python", "-m", "build", "--sdist"])
-        if rc is not None:
-            name = f"xgboost-{release}{rc}{rc_ver}.tar.gz"
-        else:
-            name = f"xgboost-{release}.tar.gz"
-        src = os.path.join(DIST, name)
-        subprocess.check_call(["twine", "check", src])
-        target = os.path.join(dist, name)
-        shutil.move(src, target)
+        subprocess.run(
+            ["patch", "-p0"], input=patch_lines, check=True, text=True, encoding="utf-8"
+        )
+
+    with DirectoryExcursion(ROOT / "python-package"):
+        subprocess.run(["python", "-m", "build", "--sdist"], check=True)
+        sdist_name = (
+            f"xgboost-{release}{rc}{rc_ver}.tar.gz"
+            if rc
+            else f"xgboost-{release}.tar.gz"
+        )
+        src = DIST / sdist_name
+        subprocess.run(["twine", "check", str(src)], check=True)
```

Comment:
is that better than `check_call`?

Reply:
According to the [documentation](https://docs.python.org/3/library/subprocess.html#using-the-subprocess-module), `run()` is now recommended:

> The recommended approach to invoking subprocesses is to use the [run()](https://docs.python.org/3/library/subprocess.html#subprocess.run) function for all use cases it can handle.

Reply:
Got it.